src/by/epam/nickgrudnitsky/entity/MarriageProblem.java

```diff
@@ -0,0 +1,88 @@
+package by.epam.nickgrudnitsky.entity;
+
+import by.epam.nickgrudnitsky.util.Validator;
+
+import java.util.*;
+
+import static by.epam.nickgrudnitsky.util.DataInput.readNumber;
+import static by.epam.nickgrudnitsky.util.Validator.*;
+
+public class MarriageProblem extends Problem {
+    private int numberOfFiances;
+
+    public MarriageProblem() {
+        setName("Marriage problem.");
+    }
+
+    private int numberOfChecks;
+    private int fianceQuality;
+    private int chosenFiance;
+    private int numberOfCorrectChoices;
+    private List<Integer> fiances = new ArrayList<>();
+
+    public void checkAlgorithm() {
+        printIntroduction();
+        numberOfFiances = readNumber();
+        numberOfChecks = readNumber();
+        if (validateNumberOfFiance(numberOfFiances) && validateNumberOfChecks(numberOfChecks)) {
+            runCheck();
+        } else {
+            printValidationFailure();
+            checkAlgorithm();
+            return;
+        }
+        printResults();
+        numberOfCorrectChoices = 0;
+    }
+
+    private void runCheck() {
+        double step = Math.round(numberOfFiances / Math.exp(1));
+        fillFiances();
+
+        for (int i = 0; i < numberOfChecks; i++) {
+
+            Collections.shuffle(fiances);
+
+            for (int j = 0; j < numberOfFiances; j++) {
+                if (fianceQuality < fiances.get(j) && j < step) {
+                    fianceQuality = fiances.get(j);
+                    continue;
+                }
+
+                if (fianceQuality < fiances.get(j)) {
+                    chosenFiance = fiances.get(j);
+                    break;
+                }
+            }
```

Comment:
Something like
`            for (int attempt = 0; attempt < numberOfFiances; attempt++) {`
                `int currentFianceQuality = fiances.get(attempt);`
`                if (bestFianceQuality < currentFianceQuality) {`
                  `  if (attempt >= stopLine) {`
                      `  chosenFiance = currentFianceQuality;`
                       ` break;`
              `      }`
                `    bestFianceQuality = currentFianceQuality;`
           `     }`
        `    }`